### ISSUE TITLE
[Feature]: Make account center request wallet permission if it's not the currently active one

### ISSUE THREAD

**Issue by ihorbond** [blocknative/web3-onboard#1225]
### Is your request related to a problem?

This request is inspired by #1219 

### Feature Description

Account Center widget shows the history of connected metamask wallets even though some of them could have been disconnected already by the user in the metamask directly and since there is no 'wallet disconnected' event we just don't know when this happens. Here's an example 
![image](https://user-images.githubusercontent.com/15573624/185758790-f457cc96-8cb9-45ca-8b80-1b2809beb133.png)
As you can see I only have ...a42d wallet connected in metamask and have since disconnected ihor.eth yet I can select it in the Account Center. 
I propose that whenever the metamask wallet is selected in the account center we send the [wallet_getPermissions](https://docs.metamask.io/guide/rpc-api.html#wallet-getpermissions) request first and look inside the caveats array. 
![image](https://user-images.githubusercontent.com/15573624/185758895-ebc77128-d75a-406b-95d5-a45b0fa7c78e.png)
If the wallet is not there anymore we send the [wallet_requestPermissions](https://docs.metamask.io/guide/rpc-api.html#wallet-requestpermissions) request so that the user can select that wallet again and make it active. We can also "clean up" other wallet addresses that are still lingering in the Account Center at this time.

### Alternative Solutions

Not that I'm aware of, I've looked all over Metamask docs.

### Anything else?

We could also send the [wallet_requestPermissions](https://docs.metamask.io/guide/rpc-api.html#wallet-requestpermissions) request whenever someone 
1) clicks on Add Account 
![image](https://user-images.githubusercontent.com/15573624/185759041-e439d928-da12-4673-9a24-04a113e89db7.png)
2) clicks "Connect another wallet" and selects Metamask from the modal window
![image](https://user-images.githubusercontent.com/15573624/185759106-5a237e33-e712-45e6-802b-19321d85f040.png)
![image](https://user-images.githubusercontent.com/15573624/185759094-305f59d0-e710-42e0-8473-043a11c39554.png)

**Comment by ihorbond** [comment]
Actually looks like just "wallet_getPermissions" won't cut it. Here's what I ended up doing whenever user selects the wallet in Account Center:

```
watchEffect(async () => {
	const { label, accounts } = connectedWallet.value 
	
	if(label === 'MetaMask') {
		
		const [primaryWallet] = accounts
		const [activeWallet] = await walletProvider.value.listAccounts()

		console.log({
			primaryWallet,
			activeWallet
		})

		if(primaryWallet !== activeWallet) {
			await walletProvider.value.send('wallet_requestPermissions', [{ eth_accounts: {} }])
		}
	}
})
```

**Comment by ihorbond** [comment]
Would it be possible to expose [this action](https://github.com/blocknative/web3-onboard/blob/5b59d100402c4e2d3a282dac8f0792b5c5d5e245/packages/core/src/store/actions.ts#L93) ? This way we could filter out the accounts[] to remove the ones that aren't connected in metamask anymore

**Comment by aaronbarnardsound** [comment]
@ihorbond thanks for these suggestions.

> Actually looks like just "wallet_getPermissions" won't cut it because even though we technically have permissions for 2 connected metamask accounts we can only send a transaction from the currently active one (right?)

I believe as long as the user has granted permission to an account, then you can send a transaction from that account (even if it is not the currently "active" one) by just setting the `from` in the transaction object.

So I think we could just check permissions when an account is selected within account center to ensure that the account permissions are still granted, and if not, then re-request them. I think only MM has the permissions rpc calls at this point, so this will not help fix other wallets.

It would be great if MM would just return an array of all accounts that are connected in the accountsChanged event rather than just the first one :(

**Comment by aaronbarnardsound** [comment]
Hey @ihorbond can try installing core version: `2.9.0-alpha.1` and see if that works as you would expect?

**Comment by aaronbarnardsound** [comment]
Released in [v2.13.0](https://github.com/blocknative/web3-onboard/releases/tag/2.13.0)
